pkg/controller/resource/app/resource.go

```diff
@@ -0,0 +1,87 @@
+package app
+
+import (
+	"context"
+
+	"github.com/giantswarm/apiextensions/pkg/apis/application/v1alpha1"
+	"github.com/giantswarm/apiextensions/pkg/clientset/versioned"
+	"github.com/giantswarm/microerror"
+	"github.com/giantswarm/micrologger"
+	"k8s.io/client-go/kubernetes"
+)
+
+const (
+	// Name is the identifier of the resource.
+	Name = "app"
+)
+
+type Config struct {
+	G8sClient versioned.Interface
+	K8sClient kubernetes.Interface
+	Logger    micrologger.Logger
+
+	GetCurrentStateFunc func(ctx context.Context, obj interface{}) ([]*v1alpha1.App, error)
+	GetDesiredStateFunc func(ctx context.Context, obj interface{}) ([]*v1alpha1.App, error)
```

Comment:
I would find it cooler to have an interface of a partial resource that the other side implements and that is injected here.

Reply:
I extracted an interface. PTALA.